Updates to replay docs for mass-replay features (#1732)
temporalio/documentation: docs-src/python/how-to-replay-a-workflow-execution-in-python.md

  - python
---

To replay Workflow Executions, use the
[`replay_workflows`](https://python.temporal.io/temporalio.worker.Replayer.html#replay_workflows)
or
[`replay_workflow`](https://python.temporal.io/temporalio.worker.Replayer.html#replay_workflow)

> methods, passing multiple or one Workflow Histories as arguments.

Author: Is this just an internal convention or what? Markdown will render these all as one line, and I find this a bit easier for editors to read. Not a big deal just curious

In the following example (which, as of server 1.18, requires advanced visibility to be enabled),
histories are downloaded from the server, then replayed. The code will raise an exception if any
replay fails.

```python
workflows = client.list_workflows(f"TaskQueue=foo and StartTime > '2022-01-01T12:00:00'")
histories = workflows.map_histories()
replayer = Replayer(
    workflows=[MyWorkflowA, MyWorkflowB, MyWorkflowC]
)
await replayer.replay_workflows(histories)
```

In the next example, a single history is loaded from a JSON string:

```python
replayer = Replayer(workflows=[YourWorkflow])
await replayer.replay_workflow(WorkflowHistory.from_json(history_json_str))
```

> In both examples if Workflow History is non-deterministic, an error will be thrown. You can choose
> to wait until all histories have been replayed with `replay_workflows` by setting the `fail_fast`
> option to `false`.

Reviewer: ```suggestion
In both examples, if Event History is non-deterministic, an error is thrown.
You can choose to wait until all histories have been replayed with `replay_workflows` by setting the `fail_fast` option to `false`.
```
